x/valset-pref/keeper.go

```diff
@@ -61,37 +61,6 @@ func (k Keeper) GetDelegationPreferences(ctx sdk.Context, delegator string) (typ
 	return valSet, nil
 }
 
-// GetValSetPreferencesWithDelegations fetches the delegator's validator set preferences
-// considering their existing delegations.
-// -If validator set preference does not exist and there are no existing delegations, it returns an error.
-// -If validator set preference exists and there are no existing delegations, it returns the existing preference.
-// -If there is any existing delegation:
-// calculates the delegator's shares in each delegation
-// as a ratio of the total shares and returns it as part of ValidatorSetPreferences.
-func (k Keeper) GetValSetPreferencesWithDelegations(ctx sdk.Context, delegator string) (types.ValidatorSetPreferences, error) {
-	delAddr, err := sdk.AccAddressFromBech32(delegator)
-	if err != nil {
-		return types.ValidatorSetPreferences{}, err
-	}
-
-	valSet, exists := k.GetValidatorSetPreference(ctx, delegator)
-	existingDelegations := k.stakingKeeper.GetDelegatorDelegations(ctx, delAddr, math.MaxUint16)
-
-	// No existing delegations for a delegator when valSet does not exist
-	if !exists && len(existingDelegations) == 0 {
-		return types.ValidatorSetPreferences{}, fmt.Errorf("No Existing delegation to unbond from")
-	}
-
-	// Returning existing valSet when there are no existing delegations
-	if exists && len(existingDelegations) == 0 {
-		return valSet, nil
-	}
-
-	// when existing delegation exists, have it based upon the existing delegation
-	// regardless of the delegator having valset pref or not
-	return types.ValidatorSetPreferences{Preferences: formatToValPrefArr(existingDelegations)}, nil
```

Comment:
Note that this would always overwrite the validator set preference with the actual delegations